src/enumMonth/Month.java

```diff
@@ -0,0 +1,76 @@
+package enumMonth;
+
+public enum Month {
+    JANUARY(31, 1, 2, 3, 6, 28),
+    FEBRUARY(29),
+    MARCH(31, 8),
+    APRIL(30),
+    MAY(31, 1, 8, 9, 28),
+    JUNE(30, 1),
+    JULY(31),
+    AUGUST(31),
+    SEPTEMBER(30, 21),
+    OCTOBER(31, 31),
+    NOVEMBER(30),
+    DECEMBER(31, 31);
+
+
+    private int[] holidays;
+    private int daysInAMonth;
+
+    Month(int daysInAMonth, int... holidays) {
+        if (daysInAMonth > 31 || daysInAMonth < 27) {
+            throw new IllegalArgumentException();
+        }
+        this.daysInAMonth = daysInAMonth;
+        this.holidays = new int[holidays.length];
+        for (int i = 0; i < holidays.length; i++) {
+            if (holidays[i] < 0 || holidays[i] > daysInAMonth) {
+                throw new IllegalArgumentException();
+            }
+            this.holidays[i] = holidays[i];
+        }
+    }
+
+    private static boolean dayExistsInHolidays(int day, Month m) {
+        for (int i = 0; i < m.holidays.length; i++) {
+            if (day == m.holidays[i]) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static void holidayCheck(String date) {
+        String[] monthAndDay = new String[2];
+        monthAndDay = date.split(" ");
+        int day = Integer.parseInt(monthAndDay[1]);
+        for (Month m : Month.values()) {
+            if (m.toString().equalsIgnoreCase(monthAndDay[0])) {
+                if (dayExistsInHolidays(day, m)) {
+                    System.out.println(String.format("%s is a public holiday", date));
+                } else {
+                    System.out.println(String.format("%s is not a public holiday", date));
+                }
+                break;
+            }
+        }
+    }
+
+    public int getDaysInAMonth() {
+        return daysInAMonth;
+    }
+
+    public int getHolidaysInAMonth() { return holidays.length;  }
```

Comment:
may be getHolidaysCountInMonth(), but it will be good to have getHolidaysInAMonth{ return holidays;}